python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -155,6 +193,13 @@ def checkExistingOutputs(self, quantum, butler, taskDef):
         collection = butler.run
         registry = butler.registry
 
+        if self.skipExisting and taskDef.metadataDatasetName is not None:
+            # Metadata output exists; this is sufficient to assume the previous
+            # run was successful and should be skipped.
+            if (ref := butler.registry.findDataset(taskDef.metadataDatasetName, quantum.dataId)) is not None:
```

Comment:
Why are we looking this up? Shouldn't this be known prior to execution time? Not if it exists, because you might be re-running the graph, but the reference should be known right?

Reply:
It's just easier to get the data ID and dataset type name separately than to find the right (unresolved) `DatasetRef` in the quantum and pass its data ID and dataset type to `findDataset`.  The behavior is identical the same either way because they have the same information content.